test/cpp/jit/test_alias_analysis.cpp

```diff
@@ -620,6 +620,31 @@ TEST(ContainerAliasingTest, PrimitveValuesDontAliasContainers) {
   }
 }
 
+TEST(ContainerAliasingTest, UnionAliasing) {
```

Comment:
Could you test the case that Elias provided as well? IIRC, it is about `List[Union[Tensor, int]]`